### PR TITLE
Unsplit improvements

### PR DESCRIPTION
**Pull request by jq-rs**
Handle empty self and other for unsplit. Change extend() to extend_from_slice().

### CODE REVIEW

**Comment by jq-rs** [comment]
Fixes #172

File: src/bytes.rs [hunk 1]
```diff
@@ -1401,6 +1401,15 @@ impl BytesMut {
     pub fn unsplit(&mut self, other: BytesMut) {
         let ptr;
 
+        if other.inner.len == 0 {
```
Comment (carllerche) [inline src/bytes.rs]: You cannot access the fields directly as they may be used to store the inline data. Instead, use the fns.

File: src/bytes.rs [hunk 2]
```diff
@@ -1401,6 +1401,15 @@ impl BytesMut {
...
+            return;
+        }
+
+        if self.inner.len == 0 {
```
Comment (carllerche) [inline src/bytes.rs]: Same here.

**Comment by jq-rs** [comment]
Thanks for the comments! Yep, an fn should be used of course. Somehow I already forgot about the inline. Now tests seem to pass just fine.

**Pull request merged**